tools/common_lib/src/gemm.h

```diff
@@ -635,42 +838,73 @@ class GemmBaseDispatcher : public NodeDispatcher
         if (params_.dt == DataType::eFp32)
         {
             randomize_linear_container_float(random_generator, uniform_distribution, input_data_a_);
-            randomize_linear_container_float(random_generator, uniform_distribution, input_data_b_);
-            randomize_linear_container_float(random_generator, uniform_distribution, input_data_c_);
+            if (params_.shape_c.get_dims_count() > 0)
+            {
+                randomize_linear_container_float(random_generator, uniform_distribution, input_data_c_);
+            }
+            randomize_linear_container_float(random_generator, uniform_distribution, input_data_dequantized_b_);
         }
         else if (params_.dt == DataType::eFp16)
         {
             randomize_linear_container_half(random_generator, uniform_distribution, input_data_a_);
-            randomize_linear_container_half(random_generator, uniform_distribution, input_data_b_);
-            randomize_linear_container_half(random_generator, uniform_distribution, input_data_c_);
+            if (params_.shape_c.get_dims_count() > 0)
+            {
+                randomize_linear_container_half(random_generator, uniform_distribution, input_data_c_);
+            }
+            randomize_linear_container_half(random_generator, uniform_distribution, input_data_dequantized_b_);
```

Comment:
We could use call this function with different range of random values for different parts of the buffer, just to add some more data variance to test quantization a bit better.

Reply:
Will do this change later as a different PR.